From 4e66154c1a13775b78e98592a2597a6d1b9cd264 Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Tue, 13 Jan 2015 14:06:41 +0100
Subject: [PATCH] Use co-installable qca-qt5 version

Without this okteta links to the Qt4 version of qca resulting on
a crash on startup.
---
 CMakeLists.txt                    |  6 ++---
 cmake/modules/FindQCA2.cmake      | 53 ---------------------------------------
 kasten/controllers/CMakeLists.txt | 16 +++++-------
 3 files changed, 9 insertions(+), 66 deletions(-)
 delete mode 100644 cmake/modules/FindQCA2.cmake

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8495f14..b85ebd4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,9 +55,9 @@ if( UNIX )
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++0x -fno-rtti")
 endif()
 
-find_package(QCA2 2.1.0)
-set_package_properties(QCA2 PROPERTIES DESCRIPTION "Qt Cryptographic Architecture"
-                       URL "https://projects.kde.org/projects/kdesupport/qca" TYPE OPTIONAL
+find_package(Qca-qt5 2.1.0)
+set_package_properties(Qca-qt5 PROPERTIES DESCRIPTION "Qt Cryptographic Architecture"
+                       URL "http:/download.kde.org/stable/qca-qt5" TYPE OPTIONAL
                        PURPOSE "Needed for most of the algorithms of the checksum tool")
 
 # control build scope
diff --git a/cmake/modules/FindQCA2.cmake b/cmake/modules/FindQCA2.cmake
deleted file mode 100644
index b9432d2..0000000
--- a/cmake/modules/FindQCA2.cmake
+++ /dev/null
@@ -1,53 +0,0 @@
-# - Try to find QCA2 (Qt Cryptography Architecture 2)
-# Once done this will define
-#
-#  QCA2_FOUND - system has QCA2
-#  QCA2_INCLUDE_DIR - the QCA2 include directory
-#  QCA2_LIBRARIES - the libraries needed to use QCA2
-#  QCA2_DEFINITIONS - Compiler switches required for using QCA2
-#
-# use pkg-config to get the directories and then use these values
-# in the FIND_PATH() and FIND_LIBRARY() calls
-
-# Copyright (c) 2006, Michael Larouche, <michael.larouche@kdemail.net>
-# Copyright (c) 2014, Friedrich W. H. Kossebau, <kossebau@kde.org>
-#
-# Redistribution and use is allowed according to the terms of the BSD license.
-# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
-
-if (NOT WIN32)
-  find_package(PkgConfig QUIET)
-  pkg_check_modules(PC_QCA2 QUIET qca2)
-  set(QCA2_DEFINITIONS ${PC_QCA2_CFLAGS_OTHER})
-endif (NOT WIN32)
-
-find_library(QCA2_LIBRARIES
-             WIN32_DEBUG_POSTFIX d
-             NAMES qca
-             HINTS ${PC_QCA2_LIBDIR} ${PC_QCA2_LIBRARY_DIRS}
-)
-
-find_path(QCA2_INCLUDE_DIR QtCrypto
-          HINTS ${PC_QCA2_INCLUDEDIR} ${PC_QCA2_INCLUDE_DIRS}
-          PATH_SUFFIXES QtCrypto
-)
-
-find_file(QCA2_VERSION_FILE
-          QtCrypto/qca_version.h
-          HINTS ${QCA2_INCLUDE_DIR}
-)
-
-if(QCA2_VERSION_FILE)
-  file(READ ${QCA2_VERSION_FILE} QCA2_VERSION_CONTENT)
-  string (REGEX MATCH "QCA_VERSION_STR \".*\"\n" QCA2_VERSION_MATCH "${QCA2_VERSION_CONTENT}")
-
-  if(QCA2_VERSION_MATCH)
-    string(REGEX REPLACE "QCA_VERSION_STR \"(.*)\"\n" "\\1" QCA2_VERSION ${QCA2_VERSION_MATCH})
-  endif(QCA2_VERSION_MATCH)
-endif(QCA2_VERSION_FILE)
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(QCA2  REQUIRED_VARS QCA2_LIBRARIES QCA2_INCLUDE_DIR
-                                        VERSION_VAR  QCA2_VERSION)
-
-mark_as_advanced(QCA2_INCLUDE_DIR QCA2_LIBRARIES)
diff --git a/kasten/controllers/CMakeLists.txt b/kasten/controllers/CMakeLists.txt
index e0fa2aa..1930b0a 100644
--- a/kasten/controllers/CMakeLists.txt
+++ b/kasten/controllers/CMakeLists.txt
@@ -32,11 +32,7 @@ include_directories(
   ${OKTETA_REL_DIR}/core
 )
 
-if( QCA2_FOUND )
-  include_directories( ${QCA2_INCLUDE_DIR} )
-endif( QCA2_FOUND )
-
-set(HAVE_QCA2 ${QCA2_FOUND})
+set(HAVE_QCA2 ${Qca-qt5_FOUND})
 configure_file( config-qca2.h.cmake  ${CMAKE_CURRENT_BINARY_DIR}/config-qca2.h )
 
 set( DOCUMENTINFOCONTROLLER_SRCS
@@ -72,11 +68,11 @@ set( STRINGSEXTRACTCONTROLLER_SRCS
   view/stringsextract/stringsextracttoolviewfactory.cpp
 )
 
-if( QCA2_FOUND )
+if( Qca-qt5_FOUND )
     set( QCA2CHECKSUM_SRCS
       view/libbytearraychecksum/algorithm/qca2bytearraychecksumalgorithm.cpp
     )
-endif( QCA2_FOUND )
+endif( Qca-qt5_FOUND )
 
 set( LIBCHECKSUM_SRCS
   view/libbytearraychecksum/abstractbytearraychecksumalgorithm.cpp
@@ -535,9 +531,9 @@ target_link_libraries( ${oktetakastencontrollers_LIB}
   Qt5::PrintSupport
   Qt5::Script
 )
-if( QCA2_FOUND )
-  target_link_libraries( ${oktetakastencontrollers_LIB} ${QCA2_LIBRARIES} )
-endif( QCA2_FOUND )
+if( Qca-qt5_FOUND )
+  target_link_libraries( ${oktetakastencontrollers_LIB} qca-qt5 )
+endif( Qca-qt5_FOUND )
 
 set_target_properties( ${oktetakastencontrollers_LIB}  PROPERTIES
   OUTPUT_NAME ${oktetakastencontrollers_LIB_NAME}
-- 
2.2.1

